src/Interpreters/Cache/IFileCachePriority.h

```diff
@@ -57,20 +36,20 @@ friend class LockedCachePriority;
     /// can only traverse the records in the low priority queue.
     class IIterator
     {
-    friend class LockedCachePriorityIterator;
     public:
         virtual ~IIterator() = default;
 
+        virtual size_t use(const CacheGuard::Lock &) = 0;
+
+        virtual std::shared_ptr<IIterator> remove(const CacheGuard::Lock &) = 0;
+
         virtual const Entry & getEntry() const = 0;
 
-    protected:
         virtual Entry & getEntry() = 0;
 
-        virtual size_t use() = 0;
+        virtual void annul() = 0;
 
-        virtual void incrementSize(ssize_t) = 0;
-
-        virtual std::shared_ptr<IIterator> remove() = 0;
+        virtual void updateSize(ssize_t size) = 0;
```

Comment:
previously it was `incrementSize`, I renamed it to `updateSize` because now `size` can be not only >= 0, so `update` seems more natural than `increment` as now it can also `decrement`.

Reply:
It's confusing when a method which increases a particular value called `update`, not `increase`

Reply:
but it does not always increase, it also decreases if the value is negative, so "increase" does not seem right, isn't it?

Reply:
Just need `int64` instead of `ssize_t`, my mistake...